common/api/core-frontend.api.md

```diff
@@ -5839,7 +5838,7 @@ export class ImageryMapLayerFormat extends MapLayerFormat {
     static createMapLayerTree(layerSettings: ImageMapLayerSettings, layerIndex: number, iModel: IModelConnection): MapLayerTileTreeReference | undefined;
 }
 
-// @internal
+// @beta
 export class ImageryMapLayerTreeReference extends MapLayerTileTreeReference {
     constructor(layerSettings: MapLayerSettings, layerIndex: number, iModel: IModelConnection);

```

Comment:
Public constructors are a backwards-compatibility footgun. If a constructor must be public, consider redefining it to take a single argument, so that you can adjust the arguments in the future without breaking the API contract.
Preferring a single `args` object over multiple arguments is a [general guideline](https://www.itwinjs.org/learning/api-support-policies/#breaking-changes) but it's particularly important for constructors because you can't deprecate and replace a constructor.

Reply:
Decided to make this constructor beta and switch to a single argument. Because `MapLayerTileTreeReference` is in the signature of `ImageryMapLayerTreeReference`, it also must be beta but its constructor will stay internal. After discussing with Michel I also promoted `MapLayerTileTreeReference` getters/setters that may be useful for public consumers